File: frontend/lib/api/agent.ts
/**
 * 🤖 IA AGENT - API endpoints para interação com o assistente de IA
 */

const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || 'http://localhost:8000';

export const agentApi = {
  /**
   * Fazer pergunta ao agente de IA
   * @param question - A pergunta a ser feita ao agente
   * @param token - Token de autenticação (opcional)
   * @returns Resposta do agente
   */
  askAgent: async (question: string, token?: string) => {
    const headers: HeadersInit = {
      'Content-Type': 'application/json',
    };

    if (token) {
      headers['Authorization'] = `Bearer ${token}`;
    }

    const response = await fetch(`${API_BASE_URL}/agent/ask-agent?question=${encodeURIComponent(question)}`, {
      method: 'POST',
      headers,
    });

    if (!response.ok) {
      return null;
    }

    return response.text();
  },
  askLlm:async (question:string,token?:string)=>{
    const headers: HeadersInit = {
      'Content-Type': 'application/json',
    };

    if (token) {
      headers['Authorization'] = `Bearer ${token}`;
    }

    const response = await fetch(`${API_BASE_URL}/agent/ask-llm?question=${encodeURIComponent(question)}`, {
      method: 'POST',
      headers,
    });

    if (!response.ok) {
      return null;
    }

    return response.text();
  }
};

File: backend/routes/agent.py
from fastapi import APIRouter
from services.agent.agent import ask_agent, ask_llm

router = APIRouter(prefix="/agent", tags=["agent"])


@router.post(
    "/ask-llm",
)
def ask_for_llm(question: str):
    return ask_llm(pergunta=question)


@router.post(
    "/ask-agent",
)
def ask_for_agent(question: str):
    """
    Endpoint padronizado para fazer perguntas ao agente de IA.

    Retorna sempre um dicionário com:
    {
        "response": "texto da resposta",
        "type": "text" | "error",
        "success": true | false
    }
    """
    result = ask_agent(pergunta=question)
    # breakpoint()
    return result
